src/ai/backend/gateway/manager.py

```diff
@@ -291,35 +298,38 @@ async def health_check(request: web.Request, params: Any) -> web.Response:
         ],
     }
     result['managers'][0].update(await host_health_check())
-    if not params['agent_ids']:
+    if not agent_ids:
         return web.json_response(result)
 
     # ## Get agent host information
     async def _agent_health_check(sess: aiohttp.ClientSession, agent_id: str) -> dict:
-        none_result: dict = {agent_id: {}}
         watcher_info = await get_watcher_info(request, agent_id)
         if not watcher_info:
-            return none_result
+            return None
         watcher_url = watcher_info['addr'] / 'health'
-        with _timeout(10.0):
-            headers = {'X-BackendAI-Watcher-Token': watcher_info['token']}
-            async with sess.get(watcher_url, headers=headers) as resp:
+        headers = {'X-BackendAI-Watcher-Token': watcher_info['token']}
+        timeout = aiohttp.ClientTimeout(total=5)
+        try:
+            async with sess.get(watcher_url, headers=headers, timeout=timeout) as resp:
                 if resp.status == 200:
                     return await resp.json()
                 else:
-                    return none_result
+                    return None
+        except asyncio.CancelledError:
+            raise
+        except asyncio.TimeoutError:
+            return None
 
     # TODO: support per-watcher ssl context
     connector = aiohttp.TCPConnector()
     async with aiohttp.ClientSession(connector=connector) as sess:
-        scheduler = await aiojobs.create_scheduler(limit=4)
-        try:
-            jobs = await asyncio.gather(*[
-                scheduler.spawn(_agent_health_check(sess, aid)) for aid in params['agent_ids']
-            ])
-            agent_infos = await asyncio.gather(*[job.wait() for job in jobs])
-        finally:
-            await scheduler.close()
+        tasks = []
+        async with TaskGroup() as tg:
+            sem = asyncio.Semaphore(10)
+            for aid in agent_ids:
+                async with sem:
+                    tasks.append(tg.create_task(_agent_health_check(sess, aid)))
```

Comment:
The semaphore must be applied inside `_agent_health_check()`!

Reply:
Thanks. Moved the semaphore inside `_agent_health_check()`